src/main/java/de/rabea/Arguments.java

```diff
@@ -4,7 +4,21 @@
 
 public class Arguments {
 
-    public HashMap<String, String> parse(String[] commandLineArguments) {
+    private String[] commandLineArguments;
+
+    public Arguments(String[] commandLineArguments) {
+        this.commandLineArguments = commandLineArguments;
+    }
+
+    public int getPort() {
+        return Integer.parseInt(parseArguments().get("-p"));
```

Comment:
I moved the computation of the port and the directory to the constructor now. I just found a bit hard to debug in the past, that's why I've stopped calling methods in the constructor. But yes, I understand that it's better to calculate it only once.